src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1463,40 +1463,38 @@ JoinRangeTableEntry(JoinExpr *joinExpr, List *dependentJobList, List *rangeTable
 				   &leftColumnNames, &leftColumnVars);
 	ExtractColumns(rightRTE, rightRangeTableId, dependentJobList,
 				   &rightColumnNames, &rightColumnVars);
-
 	joinedColumnNames = list_concat(joinedColumnNames, leftColumnNames);
-	joinedColumnVars = list_concat(joinedColumnVars, leftColumnVars);
 	joinedColumnNames = list_concat(joinedColumnNames, rightColumnNames);
+	joinedColumnVars = list_concat(joinedColumnVars, leftColumnVars);
 	joinedColumnVars = list_concat(joinedColumnVars, rightColumnVars);
 
 	rangeTableEntry->eref->colnames = joinedColumnNames;
 	rangeTableEntry->joinaliasvars = joinedColumnVars;
 
-	SetJoinRelatedColumnsCompat(rangeTableEntry, list_length(joinExpr->usingClause),
-	 leftColumnVars, rightColumnVars);
+	SetJoinRelatedColumnsCompat(rangeTableEntry,
+	leftColumnNames, rightColumnNames, leftColumnVars, rightColumnVars);
 
 	return rangeTableEntry;
 }
 
-static void SetJoinRelatedColumnsCompat(RangeTblEntry *rangeTableEntry, int joinMergedCols,
-	List *leftColumnVars, List *rightColumnVars) {
-	#if PG_VERSION_NUM >= PG_VERSION_13
+static void SetJoinRelatedColumnsCompat(RangeTblEntry *rangeTableEntry,
+		List *leftColumnNames, List *rightColumnNames, List* leftColumnVars, List* rightColumnVars) {
 
-	rangeTableEntry->joinmergedcols = joinMergedCols;
+	#if PG_VERSION_NUM >= PG_VERSION_13
 
+	/* We don't have any merged columns so set it to 0 */
+	rangeTableEntry->joinmergedcols = 0; 
 	Var* var = NULL;
-	List* joinleftcols = NIL;
+	int varId = 1;
 	foreach_ptr(var, leftColumnVars) {
-		joinleftcols = lappend_int(joinleftcols, var->varno);
-	}
-
-	List* joinrightcols = NIL;
+		rangeTableEntry->joinleftcols = lappend_int(rangeTableEntry->joinleftcols, varId);
```

Comment:
Why don't we use `var`?

Reply:
`var`s varno will not be correct because it is changed internally on postgres side with https://github.com/postgres/postgres/commit/9ce77d75c5ab094637cc4a446296dc3be6e3c221

And the commit on our side:
https://github.com/citusdata/citus/pull/3900/commits/880ff28ac22cb95096cf57e757aa97bf00835e4f

Reply:
Feel compelled to say this should be a `int numvars = list_length(leftColumnVars);for(int varid=1;varid<=numvars;varid++)` loop instead of using `foreach_ptr` with unused variable

Reply:
I thought you were asking why dont we append var's varno, yeah it makes sense to update it.